### PR TITLE
Concat data when push fails in the CFF compiler

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: test/pdfs/issue16538.pdf.link [hunk 1]
```diff
@@ -0,0 +1 @@
+https://github.com/mozilla/pdf.js/files/11706201/t.pdf
```
Comment (Snuffleupagus) [inline test/pdfs/issue16538.pdf.link:1]: Looking at the contents of that document, using Google Translate, we could perhaps just commit this PDF document in the repo and avoid the linked test-case?

**Review by Snuffleupagus — APPROVED** [review]
Does using a try-catch affect performance in any (measurable) way, since the whole purpose of PR #16292 (which is where this code was added) was to improve performance here?

**Comment by calixteman** [comment]
> Does using a try-catch affect performance in any (measurable) way, since the whole purpose of PR #16292 (which is where this code was added) was to improve performance here?

In looking in the profiler there is no difference: 2.2s spent in the worker with/without the patch vs 3.3s in just using `concat`.
I looked into the details and the time spent in `add` is almost the same.

**Pull request merged**